ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/type/TypeMapperImpl.java

```diff
@@ -46,14 +61,97 @@
  */
 public class TypeMapperImpl implements TypeMapper {
 
+    public static final String COMPONENTS_SCHEMAS = "#/components/schemas/";
+    public static final String DEFINITIONS = "#/definitions/";
+    public static final String DEFINITIONS_FIELD = "definitions";
+    public static final String JSON_SCHEMA_FIELD = "$schema";
+    public static final String JSON_SCHEMA_VERSION = "http://json-schema.org/draft-04/schema#";
     private final Components components;
     private final AdditionalData componentMapperData;
+    private final OpenAPISchema2JsonSchema converter = new OpenAPISchema2JsonSchema();
 
     public TypeMapperImpl(Components components, AdditionalData componentMapperData) {
         this.components = components;
         this.componentMapperData = componentMapperData;
     }
 
+    public TypeMapperImpl(SyntaxNodeAnalysisContext context) {
+        this.components = new Components().schemas(new HashMap<>());
+        SemanticModel semanticModel = context.semanticModel();
+        Project project = context.currentPackage().project();
+        ModuleMemberVisitor moduleMemberVisitor = extractNodesFromProject(project, semanticModel);
+        this.componentMapperData = new AdditionalData(semanticModel, moduleMemberVisitor);
+    }
+
+    public Schema getSchema(TypeSymbol typeSymbol) throws UnsupportedOperationException {
+        return getSchema(typeSymbol, true).schema();
+    }
+
+    public SchemaResult getSchema(TypeSymbol typeSymbol, boolean enableExpansion)
```

Comment:
It seems like this API is used internally. Do we need to make it public in that case ?

Reply:
If we want to get the `SchemaResult`(type schema and the relevant components) by disabling the `enableExpansion`, we need this API